### ISSUE TITLE
Memory slowly increase as training progresses

### ISSUE THREAD

**Issue by ustcwhy** [salesforce/LAVIS#377]
Thanks for your wonderful work. 

I try to pre-train instrcutBLIP from scratch on 4x4 A100.  However, the GPU memory is slowly increasing as the training progresses, which leads to OUT-OF-MEMORY after nearly 10,000 steps. I wonder if you meet the same issue? How do you resolve this? 

My yaml is:
          
run:
  runner: "runner_iter"
  accum_grad_iters: 8
  
  max_iters: 640000
  warmup_steps: 16000
  iters_per_inner_epoch: 16000
 
  task: image_text_pretrain
  # optimizer
  lr_sched: "linear_warmup_cosine_lr"
  init_lr: 1e-4
  min_lr: 1e-5
  warmup_lr: 1e-6

  weight_decay: 0.05
  batch_size_train: 12
  batch_size_eval: 12
  num_workers: 16

  seed: 42

  # amp: True
  amp: False
  resume_ckpt_path: null

  evaluate: False 
  train_splits: ["train"]

  device: "cuda"
  world_size: 4
  dist_url: "env://"
  distributed: True

**Comment by KuofengGao** [comment]
Do you solve the problem of OOM? I also encounter the OOM.
I try to empty the cuda memory every iteration but it can not work.
```
del loss, outputs
torch.cuda.empty_cache()
```

**Comment by ustcwhy** [comment]
> Do you solve the problem of OOM? I also encounter the OOM. I try to empty the cuda memory every iteration but it can not work.
> 
> ```
> del loss, outputs
> torch.cuda.empty_cache()
> ```

No, I lower the batch size...

**Comment by JinhuaLiang** [comment]
This situation is the same in my case. I am using Iterative dataset (following the LAION dataset case).

**Comment by xingenju** [comment]
Same situation for 1-stage pretrain

**Comment by JinhuaLiang** [comment]
I think this also relates to #247. Hi, @dxli94, could you please help us look at this issue? maybe sharing some insights would be of gread help!

**Comment by JinhuaLiang** [comment]
> This situation is the same in my case. I am using Iterative dataset (following the LAION dataset case).

Just come back to share something that might help. My case is attribute to the mechanism of `webdataset`. If you happen to use `webdataset` when pre-training BLIP2 and encounter CPU memory running out error, try to double check:
1. the format of input images are identical. In other word, there should not exist too big figures (e.g., 1024x728 v.s. 224x224) in the tarfile;
2. try a lower buffer size in the `webdataset.shuffle()`.

Don't know if this would help. FigureX, guys.

**Comment by gpantaz** [comment]
Hey, any updates on that? I tried #247 but it wasnt really helpful in my case. I am starting with a very small batch size (16) and the gpu memory nearly doubles after a few steps, thanks :)

**Comment by MeinhardMark** [comment]
similar situation when I'm trying to finetune on my own dataset. GPU memory usage had been slowly increasing and in the middle it went OOM.

**Comment by lsnls** [comment]
Hello! 

 Is the dataset you trained on the 13 held-in datasets mentioned in the paper? The repository doesn't seem to have a download link for all the datasets, so where did you collect the data from? And the repository doesn't seem to publish the sampling weights of the datasets.

I am looking forward to your reply!
